modules/build/src/main/scala/scala/build/preprocessing/ExtractedDirectives.scala

```diff
@@ -60,14 +63,26 @@ object ExtractedDirectives {
         case None             => None
       }
 
-      val flattened = directivesOpt.map(_.getFlattenedMap.asScala.toSeq).getOrElse(Seq.empty)
-      val strictDirectives =
-        flattened.map {
-          case (k, l) =>
-            StrictDirective(k.getPath.asScala.mkString("."), l.asScala.toSeq)
+      val strictDirectives = directivesOpt.toSeq.flatMap { directives =>
+        def toStrictValue(value: UsingValue): Seq[Value[_]] = value match {
+          case uvs: UsingValues   => uvs.values.asScala.toSeq.flatMap(toStrictValue)
+          case el: EmptyLiteral   => Seq(EmptyValue(el))
+          case sl: StringLiteral  => Seq(StringValue(sl.getValue(), sl))
+          case bl: BooleanLiteral => Seq(BooleanValue(bl.getValue(), bl))
         }
+        def toStrictDirective(ud: UsingDef) = StrictDirective(
+          ud.getKey(),
+          toStrictValue(ud.getValue()),
+          ud.getPosition().getColumn()
+        )
+
+        directives.getAst match
+          case uds: UsingDefs => uds.getUsingDefs.asScala.toSeq.map(toStrictDirective)
+          case ud: UsingDef   => Seq(toStrictDirective(ud))
+          case _              => Nil // There should be nothing else here than UsingDefs or UsingDef
+      }
 
-      Right(ExtractedDirectives(strictDirectives, directivesPositionOpt))
+      Right(ExtractedDirectives(strictDirectives.reverse, directivesPositionOpt))
```

Comment:
...wait, why `.reverse`?

Reply:
The new way of creating `StrictDirective` objects traverses the tree of using directives instead of calling `getFlattenedMap`. This is done to be able to write the start position of the key to the `StrictDirective` since with `getFlattenedMap` we lose this information (key is returned as a String with no binding to the tree nor its positions). Because of that it seems the order of returned `StrictDirective`s has been reversed, so to keep the old behaviour I used `.reverse` at the end. 

However this still changes the order slightly (probably only for `target` using directives) as can be seen in [this test output](https://github.com/VirtusLab/scala-cli/pull/2622/files#diff-dddbf7feed1e70ad65fb0129a911fb3868985b44225bce659f21134062154833)